File: Dagger-Android-Sample/src/main/java/com/gabrielsamojlo/MyApplication.java
package com.gabrielsamojlo;

import com.gabrielsamojlo.dagger_212.di.AppComponent;
import com.gabrielsamojlo.dagger_212.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class MyApplication extends DaggerApplication {

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        return appComponent;
    }

}

File: Tutorial6-2ComponentFactoryVsBuilder/src/main/java/com/smarttoolfactory/tutorial6_2componentfactoryvsbuilder/MyApplication.kt
package com.smarttoolfactory.tutorial6_2componentfactoryvsbuilder

import android.app.Application
import android.preference.PreferenceManager
import com.smarttoolfactory.tutorial6_2componentfactoryvsbuilder.di.AppComponentWithBuilder
import com.smarttoolfactory.tutorial6_2componentfactoryvsbuilder.di.AppComponentWithFactory
import com.smarttoolfactory.tutorial6_2componentfactoryvsbuilder.di.DaggerAppComponentWithBuilder
import com.smarttoolfactory.tutorial6_2componentfactoryvsbuilder.di.DaggerAppComponentWithFactory


class MyApplication : Application() {

    lateinit var appComponentWithFactory: AppComponentWithFactory
        private set

    lateinit var appComponentWithBuilder: AppComponentWithBuilder
        private set

    override fun onCreate() {
        super.onCreate()

        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)

        // AppComponent that uses Factory
        appComponentWithFactory = DaggerAppComponentWithFactory
                .factory()
                .create(this, sharedPreferences)

        // AppComponent that uses Builder
        appComponentWithBuilder = DaggerAppComponentWithBuilder
                .builder()
                .application(this)
                .sharedPreferences(sharedPreferences)
                .build()
    }


}